feature/experimental/bgp/ate_tests/bgp_always_compare_med/bgp_always_compare_med_test.go

```diff
@@ -0,0 +1,487 @@
+			capabilities[cap] = true
+		}
+		for cap, present := range capabilities {
+			if !present {
+				t.Errorf("Capability not reported: %v", cap)
+			}
+		}
+	}
+}
+
+// verifyPrefixesTelemetry confirms that the dut shows the correct numbers of installed,
+// sent and received IPv4 prefixes.
+func verifyPrefixesTelemetry(t *testing.T, dut *ondatra.DUTDevice, wantInstalled, wantSent uint32) {
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	prefixesv4 := statePath.Neighbor(ateSrc.IPv4).AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Prefixes()
+	if gotInstalled := gnmi.Get(t, dut, prefixesv4.Installed().State()); gotInstalled != wantInstalled {
+		t.Errorf("Installed prefixes mismatch: got %v, want %v", gotInstalled, wantInstalled)
+	}
+	if gotSent := gnmi.Get(t, dut, prefixesv4.Sent().State()); gotSent != wantSent {
+		t.Errorf("Sent prefixes mismatch: got %v, want %v", gotSent, wantSent)
+	}
+}
+
+type bgpNeighbor struct {
+	as         uint32
+	neighborip string
+	isV4       bool
+	peerGrp    string
+}
+
+// TestRemovePrivateAS is to Validate that private AS numbers are stripped
+// before advertisement to the eBGP neighbor.
+func TestAlwaysCompareMED(t *testing.T) {
+	// DUT configurations.
+	t.Logf("Start DUT config load.")
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+	d := &oc.Root{}
+
+	// Configure interface on the DUT.
+	t.Run("Configure DUT interfaces", func(t *testing.T) {
+		t.Logf("Start DUT interface Config.")
+		configureDUT(t, dut)
+	})
+
+	// Configure Network instance type on DUT.
+	t.Run("Configure DEFAULT network instance", func(t *testing.T) {
+		t.Log("Configure Network Instance type.")
+		dutConfNIPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut))
+		gnmi.Replace(t, dut, dutConfNIPath.Type().Config(), oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+	})
+
+	// Configure BGP+Neighbors on the DUT.
+	dutConfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	t.Run("Configure BGP Neighbors", func(t *testing.T) {
+		t.Logf("Start DUT BGP Config.")
+		gnmi.Delete(t, dut, dutConfPath.Config())
+		dutConf := bgpCreateNbr(dutAS, ateAS1, dut)
+		gnmi.Replace(t, dut, dutConfPath.Config(), dutConf)
+		fptest.LogQuery(t, "DUT BGP Config", dutConfPath.Config(), gnmi.GetConfig(t, dut, dutConfPath.Config()))
+	})
+
+	// ATE Configuration.
+	var allFlows []*ondatra.Flow
+	t.Run("Configure ATE", func(t *testing.T) {
+		t.Logf("Start ATE Config.")
+		allFlows = configureATE(t, ate)
+	})
+
+	// Verify Port Status.
+	t.Run("Verify port status on DUT", func(t *testing.T) {
+		t.Log("Verifying port status.")
+		verifyPortsUp(t, dut.Device)
+	})
+
+	t.Run("Verify BGP telemetry", func(t *testing.T) {
+		t.Log("Check BGP parameters.")
+		verifyBgpTelemetry(t, dut)
+		// Verify BGP capabilities
```

Comment:
this comment is not necessary, as it just repeats the self-explanatory method call two lines below

Reply:
Removed comment.